### ISSUE TITLE
Invalid memory address

### ISSUE THREAD

**Issue by dasJ** [joushou/serve2d#7]
After running serve2d for weeks, it crashed :) I don't know, maybe something in my setup broke it or it just has a bug :beetle:. When this crash occured, I was restarting my webserver very often (serve2d is used as a proxy in front of nginx).
Panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x514c78]
goroutine 30051 [running]:
github.com/joushou/serve2.(*Server).handle(0xc208056140, 0x7fe1ca1b3f58, 0xc20800ca80, 0x7fe1ca1b3f88, 0xc20802c060, 0x83e568, 0x0, 0x0, 0xc20800a600, 0x20, ...)
/home/janne/servepack/src/root/src/github.com/joushou/serve2/serve.go:116 +0x418
github.com/joushou/serve2.(*Server).HandleConn(0xc208056140, 0x7fe1ca1b3f88, 0xc20802c060, 0x83e568, 0x0, 0x0, 0x0, 0x0)
/home/janne/servepack/src/root/src/github.com/joushou/serve2/serve.go:213 +0x685
github.com/joushou/serve2.func·001()
/home/janne/servepack/src/root/src/github.com/joushou/serve2/serve.go:236 +0x53
created by github.com/joushou/serve2.(*Server).Serve
/home/janne/servepack/src/root/src/github.com/joushou/serve2/serve.go:237 +0x162
goroutine 1 [IO wait]:
net.(*pollDesc).Wait(0xc208010290, 0x72, 0x0, 0x0)
/usr/lib/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010290, 0x0, 0x0)
/usr/lib/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010230, 0x0, 0x7fe1ca1b2b70, 0xc20802ab90)
/usr/lib/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20802c020, 0x7fe1ca1a1000, 0x0, 0x0)
/usr/lib/go/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20802c020, 0x0, 0x0, 0x0, 0x0)
/usr/lib/go/src/net/tcpsock_posix.go:244 +0x4c
github.com/joushou/serve2.(*Server).Serve(0xc208056140, 0x7fe1ca1b2ea8, 0xc20802c020)
/home/janne/servepack/src/root/src/github.com/joushou/serve2/serve.go:230 +0x76
main.main()
/home/janne/servepack/src/root/src/github.com/joushou/serve2d/serve2d.go:324 +0xa1e
goroutine 17 [syscall, 9932 minutes, locked to thread]:
runtime.goexit()
/usr/lib/go/src/runtime/asm_amd64.s:2232 +0x1
```

**Comment by kennylevinsen** [comment]
Ah! Definitely a bug in serve2. The lib can call a logging callback if one is interested in such information. Unfortunately, I had missed a check for whether a logger was defined in a single spot, which was where it would print if a handler failed to handle the connection. That's what you hit, probably due to the constant server restart.

I have fixed it in the library, and will make a new release of serve2d soon.

**Comment by kennylevinsen** [comment]
Fixed in release v0.3.1
